apps/lib/s_cb.c

```diff
@@ -425,13 +426,21 @@ int ssl_print_tmp_key(BIO *out, SSL *s)
         return 1;
     }
 
-    BIO_puts(out, "Server Temp Key: ");
     switch (EVP_PKEY_get_id(key)) {
     case EVP_PKEY_RSA:
+        BIO_puts(out, "Server Temp Key: ");
         BIO_printf(out, "RSA, %d bits\n", EVP_PKEY_get_bits(key));
         break;
 
+    case EVP_PKEY_KEYMGMT:
+        if ((keyname = EVP_PKEY_get0_type_name(key)) != NULL) {
+            BIO_puts(out, "Server Temp Key: ");
```

Comment:
I would prefer keeping the Server Temp Key output where it was and just print `"?"` as the error case of EVP_PKEY_EC does.